nixos/modules/services/web-apps/miniflux.nix

```diff
@@ -0,0 +1,123 @@
+{ config, lib, pkgs, ... }:
+
+with lib;
+let
+  cfg = config.services.miniflux;
+
+  dbUser = "miniflux";
+  dbPassword = "miniflux";
+  dbHost = "localhost";
+  dbName = "miniflux";
+in
+
+{
+  options = {
+    services.miniflux = {
+      enable = mkEnableOption "miniflux";
+
+      user = mkOption {
+        type = types.str;
+        default = "miniflux";
+        description = ''
+          User account under which miniflux runs.
+        '';
+      };
+
+      group = mkOption {
+        type = types.str;
+        default = "miniflux";
+        description = ''
+          User group under which miniflux runs.
+        '';
+      };
+
+      listenAddress = mkOption {
+        type = types.str;
+        default = "localhost";
+        description = ''
+          The host the daemon will listen on.
+        '';
+      };
+
+      listenPort = mkOption  {
+        type = types.int;
+        default = 8080;
+        description = ''
+          The port the daemon will listen on.
+        '';
+      };
+
+      adminUser = mkOption  {
+        type = types.str;
+        default = "admin";
+        description = ''
+          Miniflux admin username.
+        '';
+      };
+
+      adminPassword = mkOption {
+        type = types.str;
+        default = "password";
+        description = ''
+          Miniflux admin password.
+        '';
+      };
+    };
+  };
+
+  config = mkIf cfg.enable {
+    services.postgresql.enable = true;
+
+    users = {
+      users.miniflux = {
```

Comment:
Does miniflux require a persistent user? I'm guessing [DynamicUser](https://www.freedesktop.org/software/systemd/man/systemd.exec.html#DynamicUser=) would be sufficient for this service.

Reply:
No, you're right. Thanks for introducing me with this systemd option.

Reply:
So i tried using `DynamicUser` at the same time I noticed that `PrivateTmp = true` can be replaced by `JoinsNamespaceOf=postgresql.service` which is better security wise. But unfortunately those two options doesn't work together as far as I tried https://github.com/systemd/systemd/issues/7253.
Which one do you think is more important?

Reply:
I went by not using `DynamicUser` since `JoinsNamespaceOf` made more sense security wise than sharing `/tmp` for a web app.